distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array_Proxy.enso

```diff
@@ -33,4 +33,4 @@ type Array_Proxy
        methods.
     from_proxy_object : Any -> Array
     from_proxy_object proxy =
-        Array_Proxy.new proxy.length proxy.at
+        Array_Proxy.new proxy.length (i -> proxy.at i)
```

Comment:
This is it? Don'w we do eta-expansion at the compiler?

Reply:
Based on [comparing the graphs](https://user-images.githubusercontent.com/26887752/206692254-37169384-b474-4b80-8dc3-ace5891c4943.png) I can speed the original case also by:
```diff
--- a/engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/function/Function.java
+++ b/engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/function/Function.java
@@ -159,7 +159,7 @@
    * @return a copy of the partially applied arguments for this function
    */
   public Object[] clonePreAppliedArguments() {
-    return preAppliedArguments.clone();
+    return preAppliedArguments;
   }
 
   /**
```
Reported as [pivotal-184001258](https://www.pivotaltracker.com/n/projects/2539304/stories/184001258).

Reply:
If I were to guess, the original would be faster.